mllib-dal/src/main/native/CorrelationImpl.cpp

```diff
@@ -149,13 +149,16 @@ static void doCorrelationDaalCompute(JNIEnv *env, jobject obj, size_t rankId,
 
 #ifdef CPU_GPU_PROFILE
 static void doCorrelationOneAPICompute(
-    JNIEnv *env, jlong pNumTabData,
+    JNIEnv *env, jlong pNumTabData, jlong numRows, jlong numClos,
     preview::spmd::communicator<preview::spmd::device_memory_access::usm> comm,
-    jobject resultObj) {
+    jobject resultObj, sycl::queue &queue) {
     logger::println(logger::INFO, "oneDAL (native): GPU compute start");
     const bool isRoot = (comm.get_rank() == ccl_root);
-    homogen_table htable =
-        *reinterpret_cast<const homogen_table *>(pNumTabData);
+    double *htableArray = reinterpret_cast<double *>(pNumTabData);
```

Comment:
could you make this section a function so that all other can reuse rather than duplicate code?